requirements.txt

```diff
@@ -32,3 +32,12 @@ jsonpatch
 
 # For validating cloud-config sections per schema definitions
 jsonschema
+
+# Used by DataSourceVMware to inspect the host's network configuration during
+# the "setup()" function.
+#
+# This allows a host that uses DHCP to bring up the network during BootLocal
+# and still participate in instance-data by gathering the network in detail at
+# runtime and merge that information into the metadata and repersist that to
+# disk.
+netifaces>=0.10.9
```

Comment:
One last question here: do we need something specific in version 0.10.9 for support here?
I ask because the versions of netifaces in Ubuntu 18.04 and 20.04 only have netifaces 0.10.4. This would block our ability to release back to earlier version of Ubuntu without additional packaging and publishing work around python3-netifaces.

Reply:
Per @akutz in IRC conversation:
Yeah, it looks like we should stick with 0.10.9. At least 0.10.5! https://github.com/al45tair/netifaces/blob/master/CHANGELOG
In 0.10.5 they fixed "Respect interface priorities when determining default gateway"
That's a huge thing for the DS
I recommend focusing post-merge on a plan to move away from netifaces or how to backport specific things?

<•blackboxsw> Chad Smith  +1 right
 
Yes I think we can definitely iterate on this netifaces dependency, the fact that Ubuntu 18.04 and 20.04 both have 10.4 probably prevents us from depending on netifaces for all datasources. We can work to make this a conditional dependency that only exists in Hirsute ++. Or better yet solve the use-cases need by VMware in the cloud-init proper network utility modules/functions so that VMware doesn't need this external dependency.

Reply:
This was discussed on IRC, and it was decided to leave as-is for now since 0.10.5 and other more recent releases fix issues with determining default gateways, something on which this DS depends. This may mean that this DS cannot be included in older Ubuntu releases, but there is a post-merge effort to try and adjust/reduce/remove the dependency on netifaces anyway.